server/app/jobs/grid_service_scheduler_worker.rb

```diff
@@ -9,58 +9,69 @@ def initialize(autostart = true)
 
   def watch
     loop do
-      self.check_deploy_queue
+      if service_deploy = self.check_deploy_queue
+        self.deploy(service_deploy)
+      end
       sleep 1
     end
   end
 
+  # Fetch deploy from queue, and return it if it should be run
+  #
+  # @return [GridServiceDeploy, nil] deploy to run
   def check_deploy_queue
     service_deploy = fetch_deploy_item
-    return unless service_deploy
+    return nil unless service_deploy
+
+    fail "deploy not pending" unless service_deploy.pending?
```

Comment:
This should be impossible, but if it did happen, it it would result in finished deploy being re-run...

* broken migration that didn't set `queued_at`
* invalid spec cases